MyApplication/app/src/main/java/com/courses/bogdan/myapplication/activities/MainActivity.java

```diff
@@ -1,100 +1,147 @@
-                Toast.makeText(MainActivity.this,"i pressed the buttoon", Toast.LENGTH_SHORT).show();
+                Toast.makeText(MainActivity.this, "i pressed the buttoon", Toast.LENGTH_SHORT).show();
             }
         });
 
-        getButton.setOnClickListener(new View.OnClickListener(){
+        getButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                textView.setText(editText.getText());
+                myView.setText(textEditor.getText());
             }
         });
 
-        colorButton.setOnClickListener(new View.OnClickListener(){
+        colorButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
                 Random randomColor = new Random();
-                red= randomColor.nextInt(256);
-                green = randomColor.nextInt(256);
-                blue = randomColor.nextInt(256);
-                layout.setBackgroundColor(Color.argb(255,red,green,blue));
+                red = randomColor.nextInt(MAX_COLOR_VALUE);
+                green = randomColor.nextInt(MAX_COLOR_VALUE);
+                blue = randomColor.nextInt(MAX_COLOR_VALUE);
+                mainlayout.setBackgroundColor(Color.argb(MAX_COLOR_VALUE - 1, red, green, blue));
 
 
             }
         });
 
+        setupFragment();
+
+        View.OnClickListener listnr=new View.OnClickListener() {
+
+            @Override
+
+            public void onClick(View v) {
+
+                Intent intent = new Intent(MainActivity.this, MyReceiver.class);
+                intent.setAction("com.example.demo.action.info");
+                sendBroadcast(intent);
+
+
+            }
+
+        };
+
+
+        fragmentButon.setOnClickListener(listnr);
+
+
     }
 
-    @Override
-    protected void onResume() {
-        super.onResume();
-        Random randomColor = new Random();
-        red= randomColor.nextInt(256);
-        green = randomColor.nextInt(256);
-        blue = randomColor.nextInt(256);
-        layout.setBackgroundColor(Color.argb(255,red,green,blue));
-        getButton.setEnabled(false);
+
+    private void setupFragment() {
+        MainFragment fragment = MainFragment.newInstance();
+        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+        transaction.add(R.id.main_frame_layout, fragment).commit();
     }
 
     @Override
     protected void onStart() {
         super.onStart();
         Random randomColor = new Random();
-        red= randomColor.nextInt(256);
+        red = randomColor.nextInt(256);
         green = randomColor.nextInt(256);
         blue = randomColor.nextInt(256);
-        layout.setBackgroundColor(Color.argb(255,red,green,blue));
+        mainlayout.setBackgroundColor(Color.argb(255, red, green, blue));
+        getButton.setEnabled(false);
+    }
 
+
+    public static MainActivity getInstace() {
```

Comment:
In Android niciodata nu incercam sa obtinem instanta unei activitati. Doar la fragmente se practica asta.